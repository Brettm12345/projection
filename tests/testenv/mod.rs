use std::path::PathBuf;

use assert_cmd::Command;
use insta::assert_debug_snapshot;
use std::ffi;
use tempfile::{tempdir, TempDir};

pub struct TestEnv {
    /// Temporary working directory.
    temp_dir: TempDir,
}

impl TestEnv {
    pub fn new() -> TestEnv {
        TestEnv {
            temp_dir: tempdir().unwrap(),
        }
    }
    pub fn assert_snapshot<I, S>(&self, args: I)
    where
        I: IntoIterator<Item = S>,
        S: AsRef<ffi::OsStr>,
    {
        assert_debug_snapshot!(Command::cargo_bin("projection")
            .unwrap()
            .arg("-d")
            .arg(&format!("{}", self.temp_dir.path().display()))
            .args(args)
            .assert())
    }
    pub fn assert_temp_dir_diff(&self) {
        assert!(dir_diff::is_different(self.test_root(), tempdir().unwrap().path()).unwrap())
    }
    /// Get the root directory for the tests.
    pub fn test_root(&self) -> PathBuf {
        self.temp_dir.path().to_path_buf()
    }
}
