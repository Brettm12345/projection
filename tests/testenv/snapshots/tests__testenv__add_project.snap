---
source: tests/testenv/mod.rs
expression: "Command::cargo_bin(\"projection\").unwrap().arg(\"-d\").arg(&format!(\"{}\", self .\n                                                                 temp_dir .\n                                                                 path () .\n                                                                 display\n                                                                 ())).args(args).assert()"
---
Assert {
    output: Output {
        status: ExitStatus(
            ExitStatus(
                0,
            ),
        ),
        stdout: "Cloning brettm12345/xmonad-config...\nSuccessfully added brettm12345/xmonad-config to projects\n",
        stderr: "",
    },
}
